### DISCUSSION TITLE
How to use the value calculated from postprocessors in the "DerivativeParsedMaterial" type "Material"

### DISCUSSION THREAD

**Discussion by wenpeng231024** [idaholab/moose#18403] [Q&A General] [answered]
Dear MOOSE group,

I use the order parameter and "ElementIntegralVariablePostprocessor" to calculate the bubble size in the Postprocessors. 

[./bubble_area]
    type = ElementIntegralVariablePostprocessor
    variable = eta
    execute_on = 'INITIAL TIMESTEP_END'
[../]

Then I try to use this calculated value in the "DerivativeParsedMaterial" type "Material". 
 
  [R_bubble]
    type = DerivativeParsedMaterial
    f_name = R_bubble
    function = 'sqrt(bubble_area/3.1415926)' 
    postprocessor_names = 'bubble_area'
    derivative_order = 1
    outputs = 'exodus'
  []

But this is some error: 

Invalid function
sqrt(bubble_area/3.1415926)

in ParsedMaterialHelper.
Syntax error: Unknown identifier

Does anyone know why this happens? It seems like "DerivativeParsedMaterial" cannot get the 'bubble_area' from the postprocessor. 

Many thanks,

Peng

**Comment by GiudGiud** [comment]
Hi Peng

Maybe it s an issue with the derivative order parameter? What derivative are you planning on having?

The syntax looks fine to me otherwise

Guillaume

**Reply by wenpeng231024** [reply]
Hi Guillaume,

Thanks for your reply. I use ParsedMaterial and try it again. But still have the same error. Maybe the "postprocessor_names = 'bubble_area'" in the "Material" do not work.

I found a similar discussion. "Use a GenericFunctionMaterial to capture the PostProcessor value and add it to ParsedMaterial using the material_property_names parameter".

https://github.com/idaholab/moose/discussions/15784

It can work using that approach! 

Peng

**Reply by dschwen** [reply]
That's not a satisfactory solution! `test/tests/materials/derivative_material_interface/postprocessors.i` shows that the `postprocessor_names` parameter is working. I'm checking out your example now.

**Comment by dschwen** [comment, marked as answer]
The input snippet above works for me. Please make sure you are not getting any warnings from your MOOSE app and double check the spelling in your input. You might want to attach a complete input file if possible.

**Reply by wenpeng231024** [reply]
I run the example test/tests/materials/derivative_material_interface/postprocessors.i. But I got the error:


Lmod is automatically replacing "python/3.8" with "vtk/8.2.0".


[31m
*** ERROR ***
Invalid function
time^2

in ParsedMaterialHelper.
Syntax error: Unknown identifier[39m

srun: Job step aborted: Waiting up to 32 seconds for job step to finish.

[31m
*** ERROR ***
Invalid function
time^2

in ParsedMaterialHelper.
Syntax error: Unknown identifier[39m

--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 1 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
slurmstepd: error: *** STEP 5345226.0 ON c0700a-s18 CANCELLED AT 2021-07-22T11:10:42 ***
srun: error: c0700a-s18: tasks 0-1: Exited with exit code 1

Is this the problem with my MOOSE app?

**Reply by dschwen** [reply]
Ok, step one is to run all the tests in MOOSE. All tests should pass. If they don't, you have a different problem entirely.

**Reply by dschwen** [reply]
What system are you on?

**Reply by wenpeng231024** [reply]
I update the MOOSE and run it again. It can work now! I think it's because of the version.

Many thanks.

Peng
